app/2.0/docs/devguide/dom-template.md

```diff
@@ -60,106 +59,183 @@ except for testing. For caveats about defining elements in the main document,
+`HTMLTemplateElement`. For example:
 
 ```js
 class MyElement extends Polymer.Element {
 
   static get template() {
-    return `<style>:host { color: blue; }</style>
+    return Polymer.html`<style>:host { color: blue; }</style>
        <h2>String template</h2>
-       <div>I've got a string template!</div>`
+       <div>I've got a string template!</div>`;
   }
 }
 customElements.define('my-element', MyElement);
 ```
 
-When using a string template, the element doesn't need to provide an `is` getter (however, the tag
-name still needs to  be  passed as the first argument to `customElements.define`).
+To enable HTML code highlighting in text editors with lit-html code highlighting functionality, you
+can declare a variable called `html` to hold the `Polymer.html` object, and return it from a `template` getter:
+
+```js
+const html = Polymer.html;
+
+class MyElement extends Polymer.Element {
+
+  static get template() {
+    return html`<style>:host { color: blue; }</style>
+       <h2>String template</h2>
+       <div>I've got a string template!</div>`;
+  }
+}
+customElements.define('my-element', MyElement);
+```
+
+When using a static `template` getter, the element doesn't need to provide an `is` getter.
+However, the tag name still needs to  be  passed as the first argument to
+`customElements.define`.
 
 ### Inherited templates {#inherited-templates}
 
-An element that extends another Polymer element can inherit its template. If the element doesn't
-provide its own DOM template (using either a `<dom-module>` or a string template), Polymer uses the
-same template as the superclass, if any.
+You can inherit an existing template by extending a base class with a template. This
+functionality provides for a variety of scenarios: 
 
-You can also modify a superclass template by defining a `template` getter that returns a modified
-template element. If you're going to modify the superclass template, there are a couple of
-important rules:
+* To [inherit a base class template with **no modifications**](#nomods), do not supply a
+  template definition in the child class declaration.
+* To [**override** a template definition](#overridebase), supply your own template definition 
+  in the child class.
+* To [**extend** a base class template](#extendbase), supply a template in the child class that includes
+  a reference to the base class template.
+* To [**insert** content from a child class into placeholders supplied by a base class](#insertcontent),
+  create placeholders in the base class and supply the content with functions.
 
--   Don't modify the superclass template in place; make a copy before modifying.
--   If you're doing anything expensive, like copying or modifying an existing template,
-    you should memoize the modified template so you don't have to regenerate it when the
-    getter is called.
+#### Inherit a base class template with no modifications {#nomods}
```

Comment:
These examples should use dom-module, since it's the default mechanism for Polymer 2.x.

Reply:
Converted back to dom-module examples - a couple of issues with the following

http://plnkr.co/edit/f5utiX?p=preview
http://plnkr.co/edit/YQ5t5I?p=preview

@arthurevans @justinfagnani

Reply:
I meant that specifically for the "inherit a base class with no modifications."

The `${expr}` format is specific to template literals (those things in ``` `backticks` ```) and won't work in `<template>` elements... despite the names, they're rather different creatures.

The extension examples you added with the `html` function and template literals are much clearer than the previous ones that require manipulating template elements in JS. And for that reason I think we might want to keep them in. However, they'll only work in 2.4+, so we might want to leave in at least one of the previous examples.

I had some more specific comments on this stuff inline.